sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,247 +19,367 @@
+https://github.com/intel/llvm/issues
 
-## Feature test macro
+== Dependencies
 
-This extension provides a feature-test macro as described in the core SYCL
-specification section 6.3.3 "Feature test macros".  Therefore, an
-implementation supporting this extension must predefine the macro
-`SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the table below.
-Applications can test for the existence of this macro to determine if the
-implementation supports this feature, or applications can test the macro's
-value to determine which of the extension's APIs the implementation supports.
-
-[frame="none",options="header"]
-|======================
-|Value |Description
-|1     |The APIs of this experimental extension are not versioned, so the feature-test macro always has this value. 
-|======================
-
-## Matrix API Versions
-
-While this document presents the core API that unifies Intel AMX, Intel XMX, and Nvidia Tensor Cores, the implementations support slightly different versions of the API. For this reason, we introduce a new macro, namely `SYCL_EXT_ONEAPI_MATRIX_VERSION`  to distinguish between these different implementations. The goal in the next few months is to get rid of this implementation versioning macro. These are the current values for this macro.
+This extension is written against the SYCL 2020 revision 6 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
 
-[frame="none",options="header"]
-|======================
-|Value |Description
-|1     |Initial extension JIT implementation on Intel AMX and Intel XMX. load, store, mad, fill, piece-wise operations, and the query interface are supported. The old API used for this implementation is detailed in link:../../deprecated/sycl_ext_oneapi_matrix_no_use.asciidoc[matrix extension]
-|2     |JIT implementation on Intel AMX and Intel XMX. load, store, mad, fill, piece-wise operations, and the query interface are supported 
-|3     |Implementation on Nvidia Tensor Cores
-|======================
+== Status
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+== Backend support status
+This extension is currently implemented in {dpcpp} only for devices
+that contain a matrix hardware, specifically Intel(R) Advanced Matrix
+Extensions (Intel(R) AMX), Intel(R) Xe Matrix Extensions (Intel(R)
+XMX) and Nvidia(R) Tensor Cores.
+
+The `joint_matrix_mad` function is an optional kernel feature as defined
```

Comment:
Yes, it is accurate. We cannot detect whether the kernel is supported or not just by looking at joint matrix types individually. "joint_matrix_mad" is the one that include the 3 matrices so we can have the full used combination (sizes and types) from there.

Reply:
It seems like both the `joint_matrix` type and the `joint_matrix_mad` function are "optional kernel features".  Imagine a kernel that creates a `joint_matrix` type and does *not* call `joint_matrix_mad` at all.  Is such a kernel guaranteed to run on all possible devices?  I'm guessing that it will probably not.

I think there should probably be two rules to determine whether a kernel is supported on a device:

* Every `joint_matrix` type declaration must be one of the supported types according to the device's query, and
* Every `joint_matrix_mad` function call must use a combination of types that is supported according to the device's query.

If you agree, I think the wording can be tweaked like:

> The `joint_matrix` type and the `joint_matrix_mad` function are optional kernel features as defined ...

and:

> If the application submits a kernel using an unsupported `joint_matrix` type or calls `joint_matrix_mad` with an unsupported combination, ...